src/commands/force/user/permsetlicense/assign.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as os from 'os';
+import { flags, FlagsConfig, SfdxCommand } from '@salesforce/command';
+import { Aliases, Messages, SfdxError, User, UserFields } from '@salesforce/core';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-user', 'permsetlicense.assign');
+
+type SuccessMsg = {
+  name: string;
+  value: string;
+};
+
+type FailureMsg = {
+  name: string;
+  message: string;
+};
+
+export type PSLResult = {
+  successes: SuccessMsg[];
+  failures: FailureMsg[];
+};
+
+interface PermissionSetLicense {
+  Id: string;
+}
+
+export class UserPermsetLicenseAssignCommand extends SfdxCommand {
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessage('examples').split(os.EOL);
+  public static readonly requiresUsername = true;
+  public static readonly flagsConfig: FlagsConfig = {
+    name: flags.string({
+      char: 'n',
+      description: messages.getMessage('flags.name'),
+      required: true,
+    }),
+    onbehalfof: flags.array({
+      char: 'o',
+      description: messages.getMessage('flags.onBehalfOf'),
+    }),
+  };
+  private readonly successes: SuccessMsg[] = [];
+  private readonly failures: FailureMsg[] = [];
+  private pslId: string;
+
+  public async run(): Promise<PSLResult> {
+    const usernames = (this.flags.onbehalfof as string[]) ?? [this.org.getUsername()];
+    this.logger.debug(`will assign permset to users: ${usernames.join(', ')}`);
+    const pslName = this.flags.name as string;
+
+    const conn = this.org.getConnection();
+    try {
+      this.pslId = (
+        await conn.singleRecordQuery<PermissionSetLicense>(
+          `select Id from PermissionSetLicense where DeveloperName = '${pslName}' or MasterLabel = '${pslName}'`
+        )
+      ).Id;
+    } catch {
+      throw new SfdxError('PermissionSetLicense not found');
+    }
+    (await Promise.all(usernames.map((username) => this.usernameToPSLAssignment({ pslName, username })))).map(
+      (result) => {
+        if (isSuccess(result)) {
+          this.successes.push(result);
+        } else {
+          this.failures.push(result);
+        }
+      }
+    );
+
+    this.print();
+    this.setExitCode();
+
+    return {
+      successes: this.successes,
+      failures: this.failures,
+    };
+  }
+
+  // handles one username/psl combo so these can run in parallel
+  private async usernameToPSLAssignment({
+    pslName,
+    username,
+  }: {
+    pslName: string;
+    username: string;
+  }): Promise<SuccessMsg | FailureMsg> {
+    // Convert any aliases to usernames
+    const aliasOrUsername = (await Aliases.fetch(username)) || username;
```

Comment:
Isn't the naming backwards here? The `username` var could be either an Alias or username, while after L96 we've converted any aliases to usernames, or kept the username. I think the parameter should be `aliasOrUsername` while L96 becomes `const username = (await Aliases.fetch(aliasOrUsername)) || aliasOrUsername;`

Reply:
yeah, I borrowed a lot of this from permsetAssign.  fixing.